src/electionguard/ballot.cpp

```diff
@@ -328,13 +328,14 @@ namespace electionguard
         }
 
         // need to make sure we use the nonce used in precomputed values
-        auto nonce = precomputedTwoTriplesAndAQuad->get_triple1()->get_exp();
+        auto nonce = precomputedTwoTriplesAndAQuad->get_triple1()->clone_exp();
```

Comment:
left this as a clone. it might not be necessary but this one i left to minimize impact